provider/adapters/src/any_payload.rs

```diff
@@ -71,6 +71,25 @@ impl AnyPayloadProvider {
         }
     }
 
+    /// Creates an `AnyPayloadProvider` from an existing [`DataPayload`].
+    pub fn from_payload<M: KeyedDataMarker + 'static>(payload: DataPayload<M>) -> Self
```

Comment:
These names are inconsistent with the other constructors, but `new_` doesn't work here, so please rename the others to `from_`